### DISCUSSION TITLE
Trouble With AssumeRoleWithCertificate

### DISCUSSION THREAD

**Discussion by haylch** [minio/minio#19916] [Q&A] [answered]
Anyone can help? I have issues enabling AssumeRoleWithCertificate and getting it to work.
Have read through the doc 
https://github.com/minio/minio/blob/master/docs/sts/tls.md
and have done the steps indicated in the doc as follows:

- Enable TLS
mc admin config set myminio identity_tls skip_verify=off

- Restart endpoint for alias myminio
mc admin service restart myminio

- Add a policy with name which is the SAME as the CN in the client cert e.g minioclient
mc admin policy create myminio minioclient /mnt/Shared/yaml/minioclient_policy.json

	minioclient_policy.json
	{
	  "Version": "2012-10-17",
	  "Statement": [
		{
		  "Action": [
		  "s3:*"
		  ],
		  "Effect": "Allow",
		  "Resource": ["arn:aws:s3:::*"]
		}
	  ]
	}

- Generated client cert contains the Extended Key Usage:
TLS Web Client Authentication

But executing the following command:
e.g curl --insecure -X POST --key private.key --cert public.crt "https://minio:30000?Action=AssumeRoleWithCertificate&Version=2011-06-15&DurationSeconds=3600"

I still received response saying STS API AssumeRoleWithCerificate is disabled.

What steps am I missing here? Help please. I'm running MinIo version 5.0.14 on kubernetes RKE2 on ubuntu 22.04.

**Comment by harshavardhana** [comment, marked as answer]
Read my comment https://github.com/minio/minio/issues/19917
